### PR TITLE
Changed Table and DataTable to improve theme customization

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed Table and DataTable to improve theme customization

#### Where should the reviewer start?

base.js

#### What testing has been done on this PR?

storybook

#### How should this be manually tested?

storybook

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/2336
https://github.com/grommet/grommet/issues/2092

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/Cell.js [hunk 1]
```diff
@@ -21,30 +19,16 @@ export const Cell = ({
 
   if (typeof content === 'string' || typeof content === 'number') {
     if (primary) {
-      content = <strong>{content}</strong>;
+      content = <Text><strong>{content}</strong></Text>;
```
Comment (alansouzati) [inline src/js/components/DataTable/Cell.js]: should the strong here move removed and replaced with a theme entry for weight?

File: src/js/components/Table/StyledTable.js [hunk 2]
```diff
@@ -22,19 +22,19 @@ export const StyledTableCell = styled.td`
   padding: 0;
   font-weight: inherit;
   text-align: inherit;
-  height: inherit;
+  height: 100%;
```
Comment (alansouzati) [inline src/js/components/Table/StyledTable.js]: why are we changing this?

File: src/js/components/Table/StyledTable.js [hunk 3]
```diff
@@ -22,19 +22,19 @@ export const StyledTableCell = styled.td`
...
-  ${props => !props.verticalAlign && props.tableContext === 'footer' && 'vertical-align: top;'}
+  ${props => props.tableContextTheme && props.tableContextTheme.extend}
 `;
 
 export const StyledTableDataCaption = styled.caption`
   display: none;
 `;
 
 export const StyledTableRow = styled.tr`
+  height: 100%;
```
Comment (alansouzati) [inline src/js/components/Table/StyledTable.js]: why?

File: src/js/components/Table/StyledTable.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Table/StyledTable.js]: It's necessary for cell backgrounds to take the full row height.

File: src/js/components/Table/StyledTable.js (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/Table/StyledTable.js]: same as previous comment

File: src/js/components/DataTable/Cell.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/Cell.js]: Agreed. Updated.

**Pull request merged**